### ISSUE TITLE
ValueError: Missing some output keys: {'text'} even if the chain finished.

### ISSUE THREAD

**Issue by Yanxin-Ye** [langchain-ai/langchain#17816]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.

### Example Code

```
from langchain.chains import SequentialChain
from langchain.chains.router import MultiPromptChain
from langchain.chains.router.llm_router import LLMRouterChain,RouterOutputParser
from langchain.prompts import PromptTemplate, ChatPromptTemplate
from langchain.chains import LLMChain
from langchain.chat_models import ChatVertexAI
from langchain.chains.router.multi_prompt_prompt import MULTI_PROMPT_ROUTER_TEMPLATE

chat=ChatVertexAI(
    model_name="chat-bison@001",
    max_output_tokens=1000,
    temperature=0.0,
    top_k=10,
    top_p=0.8
)

def create_sequential_chain(llm):
    first_prompt = ChatPromptTemplate.from_template(
        "Translate the following review to french:"
        "\n\n{input}"
    )
    # chain 1: input= Review and output= English_Review
    chain_one = LLMChain(llm=llm, prompt=first_prompt,
                         output_key="French_Review"
                        )
    second_prompt = ChatPromptTemplate.from_template(
        "Can you summarize the following review in 1 sentence:"
        "\n\n{French_Review}"
    )
    # chain 2: input= English_Review and output= summary
    chain_two = LLMChain(llm=llm, prompt=second_prompt,
                         output_key="summary"
                        )
    # prompt template 3: translate to english
    third_prompt = ChatPromptTemplate.from_template(
        "What language is the following review:\n\n{input}"
    )
    # chain 3: input= Review and output= language
    chain_three = LLMChain(llm=llm, prompt=third_prompt,
                           output_key="language"
                          )
    # prompt template 4: follow up message
    fourth_prompt = ChatPromptTemplate.from_template(
        "Write a follow up response to the following "
        "summary in the specified language:"
        "\n\nSummary: {summary}\n\nLanguage: {language}"
    )
    # chain 4: input= summary, language and output= followup_message
    chain_four = LLMChain(llm=llm, prompt=fourth_prompt,
                          output_key="followup_message"
                         )
    # overall_chain: input= Review
    # and output= English_Review,summary, followup_message
    sequential_chain = SequentialChain(
        chains=[chain_one, chain_two, chain_three, chain_four],
        input_variables=["input"],
        output_variables=["French_Review", "summary","followup_message"],
        verbose=True
    )
    return sequential_chain

sequential_chain = create_sequential_chain(chat)
####### The sequeential chain works fine
sequential_chain_response = sequential_chain("I find the taste mediocre. The foam doesn't hold, it's strange. I buy the same ones in stores and the taste is much better.")
print(sequential_chain_response)
######## Doesn't work in the router chain
analysis_template = """You are a data analyst. You are great at analyze and summarize product reviews.
When you don't know the answer to a question you admit that you don't know. \nFor example: 
I find the taste mediocre. The foam doesn't hold, it's strange.
""\nHere is a question:
{input}"""

prompt_infos = [
            {
                "name": "analysis",
                "description": "great at analyze and summarize product reviews",
                "prompt_template": analysis_template,
                "destination_chain": sequential_chain
            }
]

destination_chains = {}
for p_info in prompt_infos:
    name = p_info["name"]
    destination_chains[name] = p_info["destination_chain"]
destinations = [f"{p['name']}: {p['description']}" for p in prompt_infos]
destinations_str = "\n".join(destinations)

default_prompt = ChatPromptTemplate.from_template("{input}")
default_chain = LLMChain(llm=chat, prompt=default_prompt)

router_template = MULTI_PROMPT_ROUTER_TEMPLATE.format(
    destinations=destinations_str
)
router_prompt = PromptTemplate(
    template=router_template,
    input_variables=["input"],
    output_parser=RouterOutputParser(),
)
router_chain = LLMRouterChain.from_llm(chat, router_prompt)
chain = MultiPromptChain(router_chain=router_chain,
                         destination_chains=destination_chains,
                         default_chain=default_chain,
                         verbose=True
                         )
chain.run("I find the taste mediocre. The foam doesn't hold, it's strange. I buy the same ones in stores and the taste is much better.")
```

### Error Message and Stack Trace (if applicable)

> Entering new MultiPromptChain chain...
/opt/conda/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:117: LangChainDeprecationWarning: The function `run` was deprecated in LangChain 0.1.0 and will be removed in 0.2.0. Use invoke instead.
  warn_deprecated(
/opt/conda/lib/python3.10/site-packages/langchain/chains/llm.py:316: UserWarning: The predict_and_parse method is deprecated, instead pass an output parser directly to LLMChain.
  warnings.warn(
analysis: {'input': "I find the taste mediocre. The foam doesn't hold, it's strange. I buy the same ones in stores and the taste is much better."}

> Entering new SequentialChain chain...

> Finished chain.

> Finished chain.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[1], line 112
    106 router_chain = LLMRouterChain.from_llm(chat, router_prompt)
    107 chain = MultiPromptChain(router_chain=router_chain,
    108                          destination_chains=destination_chains,
    109                          default_chain=default_chain,
    110                          verbose=True
    111                          )
--> 112 chain.run("I find the taste mediocre. The foam doesn't hold, it's strange. I buy the same ones in stores and the taste is much better.")

File /opt/conda/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:145, in deprecated.<locals>.deprecate.<locals>.warning_emitting_wrapper(*args, **kwargs)
    143     warned = True
    144     emit_warning()
--> 145 return wrapped(*args, **kwargs)

File /opt/conda/lib/python3.10/site-packages/langchain/chains/base.py:538, in Chain.run(self, callbacks, tags, metadata, *args, **kwargs)
    536     if len(args) != 1:
    537         raise ValueError("`run` supports only one positional argument.")
--> 538     return self(args[0], callbacks=callbacks, tags=tags, metadata=metadata)[
    539         _output_key
    540     ]
    542 if kwargs and not args:
    543     return self(kwargs, callbacks=callbacks, tags=tags, metadata=metadata)[
    544         _output_key
    545     ]

File /opt/conda/lib/python3.10/site-packages/langchain_core/_api/deprecation.py:145, in deprecated.<locals>.deprecate.<locals>.warning_emitting_wrapper(*args, **kwargs)
    143     warned = True
    144     emit_warning()
--> 145 return wrapped(*args, **kwargs)

File /opt/conda/lib/python3.10/site-packages/langchain/chains/base.py:363, in Chain.__call__(self, inputs, return_only_outputs, callbacks, tags, metadata, run_name, include_run_info)
    331 """Execute the chain.
    332 
    333 Args:
   (...)
    354         `Chain.output_keys`.
    355 """
    356 config = {
    357     "callbacks": callbacks,
    358     "tags": tags,
    359     "metadata": metadata,
    360     "run_name": run_name,
    361 }
--> 363 return self.invoke(
    364     inputs,
    365     cast(RunnableConfig, {k: v for k, v in config.items() if v is not None}),
    366     return_only_outputs=return_only_outputs,
    367     include_run_info=include_run_info,
    368 )

File /opt/conda/lib/python3.10/site-packages/langchain/chains/base.py:164, in Chain.invoke(self, input, config, **kwargs)
    162     raise e
    163 run_manager.on_chain_end(outputs)
--> 164 final_outputs: Dict[str, Any] = self.prep_outputs(
    165     inputs, outputs, return_only_outputs
    166 )
    167 if include_run_info:
    168     final_outputs[RUN_KEY] = RunInfo(run_id=run_manager.run_id)

File /opt/conda/lib/python3.10/site-packages/langchain/chains/base.py:438, in Chain.prep_outputs(self, inputs, outputs, return_only_outputs)
    420 def prep_outputs(
    421     self,
    422     inputs: Dict[str, str],
    423     outputs: Dict[str, str],
    424     return_only_outputs: bool = False,
    425 ) -> Dict[str, str]:
    426     """Validate and prepare chain outputs, and save info about this run to memory.
    427 
    428     Args:
   (...)
    436         A dict of the final chain outputs.
    437     """
--> 438     self._validate_outputs(outputs)
    439     if self.memory is not None:
    440         self.memory.save_context(inputs, outputs)

File /opt/conda/lib/python3.10/site-packages/langchain/chains/base.py:269, in Chain._validate_outputs(self, outputs)
    267 missing_keys = set(self.output_keys).difference(outputs)
    268 if missing_keys:
--> 269     raise ValueError(f"Missing some output keys: {missing_keys}")

ValueError: Missing some output keys: {'text'}

### Description

I am trying to use a router chain to route human input to a sequential chain or a default llm. 
Given the same prompt, the sequential chain worked fine, but the router chain returned "ValueError: Missing some output keys: {'text'}". From the output I can see the error occured after the chain finished, and I don't have any variable named "text". Can you take a look at this? Thank you!

### System Info

System Information
------------------
> OS:  Linux
> OS Version:  #1 SMP Debian 5.10.209-2 (2024-01-31)
> Python Version:  3.10.12 | packaged by conda-forge | (main, Jun 23 2023, 22:40:32) [GCC 12.3.0]

Package Information
-------------------
> langchain_core: 0.1.23
> langchain: 0.1.7
> langchain_community: 0.0.20
> langsmith: 0.0.87
> langchain_google_vertexai: 0.0.5
> langchainplus_sdk: 0.0.20

Packages not installed (Not Necessarily a Problem)
--------------------------------------------------
The following packages were not found:

> langgraph
> langserve

**Comment by Yanxin-Ye** [comment]
Thank you bot. But I never specified any output key as "text" and I don't need one. How come it expects an output key named "text" ? I looked into RouterOutputParser() and it doesn't populate any output key "text".

**Comment by Yanxin-Ye** [comment]
I think the issue comes from this line:
https://github.com/langchain-ai/langchain/blob/c1bb5fd498427bef3749dbba41d2483148f53f35/libs/langchain/langchain/chains/router/multi_prompt.py#L22

Why a "text" key is required here? My sequential chain return multiple output and I think we should be able to specify which output should go to the final answer.

**Comment by Yanxin-Ye** [comment]
This issue can be solved by adding another llm chain to the last step of sequential chain.

**Comment by liyangwd** [comment]
@Yanxin-Ye   This issue can be solved by adding another llm chain to the last step of sequential chain.  can you share this?

**Comment by liyangwd** [comment]
when i connect MultiPromptChain with RetrievalQAWithSourcesChain i rewrite output_keys to fix this 
`class SelfMultiPromptChain(MultiPromptChain):
    @property
    def output_keys(self) -> List[str]:
        return ["answer"]`
